### PR TITLE
Add CLI commands for app versioning

### PR DESCRIPTION
**Pull request by ecfm**
http://builds.cask.co/browse/CDAP-DUT4963-4

### CODE REVIEW

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/app/ListAppVersionsCommand.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    String appName = arguments.get(ArgumentName.APP.toString());
+    List<String> versions = appClient.listAppVersions(cliConfig.getCurrentNamespace(), appName);
+
+    Table table = Table.builder()
+      .setHeader("version")
+      .setRows(versions, new RowMaker<String>() {
+        @Override
+        public List<?> makeRow(String object) {
```
Comment (anwar6953) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/app/ListAppVersionsCommand.java]: this is parameterized, so can make the input object something other than Object, right?

File: cdap-cli/src/main/java/co/cask/cdap/cli/util/AbstractAuthCommand.java [hunk 2]
```diff
@@ -16,23 +16,32 @@
...
 import java.io.PrintStream;
+import java.util.Arrays;
 import java.util.Set;
 
 /**
  * Abstract command for updating {@link co.cask.cdap.security.authentication.client.AccessToken}.
  */
 public abstract class AbstractAuthCommand implements Command {
 
+  private static Joiner joiner = Joiner.on(".");
```
Comment (anwar6953) [inline cdap-cli/src/main/java/co/cask/cdap/cli/util/AbstractAuthCommand.java]: rename it dotJoiner, would be better

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/app/ListAppVersionsCommand.java (hunk 1 above)
Comment (ecfm) [reply cdap-cli/src/main/java/co/cask/cdap/cli/command/app/ListAppVersionsCommand.java]: Which is parameterized? The input is just String of version, should have renamed it as version.

Comment (anwar6953) [reply cdap-cli/src/main/java/co/cask/cdap/cli/command/app/ListAppVersionsCommand.java]: the makeRow method's parameter is parameterizable.

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/app/ListAppVersionsCommand.java [hunk 3]
```diff
@@ -53,7 +53,7 @@ public void perform(Arguments arguments, PrintStream output) throws Exception {
       .setHeader("version")
       .setRows(versions, new RowMaker<String>() {
         @Override
-        public List<?> makeRow(String version) {
+        public List makeRow(String version) {
```
Comment (anwar6953) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/app/ListAppVersionsCommand.java]: can make it List<String>, right?

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/StartProgramCommand.java [hunk 4]
```diff
@@ -56,23 +56,24 @@ public void perform(Arguments arguments, PrintStream output) throws Exception {
...
 
-    String runtimeArgsString = arguments.get(ArgumentName.RUNTIME_ARGS.toString(), "");
+    String runtimeArgsString = arguments.getOptional(ArgumentName.RUNTIME_ARGS.toString(), "");
     if (runtimeArgsString == null || runtimeArgsString.isEmpty()) {
       // run with stored runtime args
-      programClient.start(programId.toId(), isDebug);
-      runtimeArgsString = GSON.toJson(programClient.getRuntimeArgs(programId.toId()));
-      output.printf("Successfully started %s '%s' of application '%s' with stored runtime arguments '%s'\n",
-                    elementType.getName(), programName, appId, runtimeArgsString);
+      programClient.start(programId, isDebug, null);
```
Comment (anwar6953) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/StartProgramCommand.java:67]: no need to pass in null as 3rd parameter, right?

Comment (ecfm) [reply cdap-cli/src/main/java/co/cask/cdap/cli/command/StartProgramCommand.java:67]: This is currently the only method that accepts ProgramId to start a program, may add more methods to start a program in the future.

Comment (anwar6953) [reply cdap-cli/src/main/java/co/cask/cdap/cli/command/StartProgramCommand.java:67]: ah ok.

**Comment by anwar6953** [comment]
LGTM.
Do squash commits.

**Comment by ecfm** [comment]
Build is green http://builds.cask.co/browse/CDAP-DUT4963-4

**Pull request merged**